### PR TITLE
library ingestions into graph

### PR DESCRIPTION
**Pull request by yingjieMiao**
@stkem @leogrim @andrewconner 
- create/remove keep-lib, user-lib edges
- created `NeighborQuerier` trait to enable easy nb query of the graph
- farily comprehensive test for graph data ingestion

### CODE REVIEW

**Comment by leogrim** [comment]
:+1: for the ingestion part
Check out GraphPrimitives for the query part, it'd be nice to build primitives in there rather than tying them to random walks. We may want to use them in a different context :)

**Comment by yingjieMiao** [comment]
oh, nice. I didn't know it was there.

File: kifi-backend/modules/graph/app/com/keepit/graph/manager/GraphUpdater.scala [hunk 1]
```diff
@@ -112,8 +117,11 @@ class GraphUpdaterImpl @Inject() () extends GraphUpdater {
         writer.saveEdge(keepVertexId, userVertexId, EmptyEdgeData)
         writer.saveEdge(keepVertexId, uriVertexId, EmptyEdgeData)
 
-      case _ =>
-        writer.removeVertexIfExists(keepVertexId)
+        writer.saveEdge(libVertexId, keepVertexId, EmptyEdgeData)
+        writer.saveEdge(keepVertexId, libVertexId, EmptyEdgeData)
+
+      case _ => // do nothing
```
Comment (leogrim) [inline kifi-backend/modules/graph/app/com/keepit/graph/manager/GraphUpdater.scala:124]: We need to remove the keep vertex when it's not active don't we?

**Comment by leogrim** [comment]
We may also want to ingest library vertex from the library table, so that vertices corresponding to inactive libraries can be cleared out.

File: kifi-backend/modules/graph/app/com/keepit/graph/manager/GraphUpdater.scala (hunk 1 above)
Comment (yingjieMiao) [inline kifi-backend/modules/graph/app/com/keepit/graph/manager/GraphUpdater.scala:124]: it's removed before the `match`

Comment (leogrim) [inline kifi-backend/modules/graph/app/com/keepit/graph/manager/GraphUpdater.scala:124]: Oh right, sorry :)

**Comment by yingjieMiao** [comment]
@leogrim  I think you are right. will ingest that part.

**Pull request merged**